### DISCUSSION TITLE
开发者例会议题收集（20220920）

### DISCUSSION THREAD

**Discussion by chuntaojun** [polarismesh/polaris#655] [Meeting]
# 开发者例会议题收集（20220920）

## 会议信息

andrewshan(单家骏) 邀请您参加腾讯会议
会议主题：Polaris社区开发者双周会
会议时间：2022/09/20 19:30-20:00 (GMT+08:00) 中国标准时间 - 北京
重复周期：2022/07/12-2022/10/04 19:30-20:00, 每两周 (周二)

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/kBemx9IzxmGE

#腾讯会议：701-1815-5065

复制该信息，打开手机腾讯会议即可参与

**Comment by chuntaojun** [comment]
## 北极星SDK获取所处地理位置信息流程

背景：就近路由

新的SDK获取所处地理位置信息流程，将由两个插件组成插件链：local -> remote

- local插件：直接配置 region、zone、campus 的信息，该信息配置在 polaris.yaml 配置，支持环境变量占位符
- remoteHttp插件：目前仅支持 http 协议进行获取，该信息配置在 polaris.yaml 中，region、zone、campus 各自配置自己的远程调用请求, 响应结果直接作为 region、zone、campus 的值
- remoteService插件：根据北极星服务端的ReportClient接口实现一个grpc服务，用于SDK上报获取地理位置信息


```yaml
#描述:全局配置项
global:
  location:
    providers:
    - name: xxx
      type: local
      region: ${REGION}
      zone: ${ZONE}
      campus: ${CAMPUS}
    - name: xxx
      type: remoteHttp
      region: http://127.0.0.1/region
      zone: http://127.0.0.1/zone
      campus: http://127.0.0.1/campus
    - name: xxx
      type: remoteService
      address: grpc://xxx

**Reply by chuntaojun** [reply]
remote 插件，response object 的格式如果 json 的话，走固定的 pb 结构定义

**Reply by chuntaojun** [reply]
local -> remoteText(需要手动开启) -> remoteService(需要手动开启)

**Comment by wallezhang** [comment]
同一个SDK注册的多个服务实例，心跳包可以只维护一个，只需要确保服务是存活的即可，减少网络和资源消耗

**Reply by chuntaojun** [reply]
// 可实现，待社区同学设计

**Comment by onecer** [comment]
Polaris-server Swagger建设进度汇报

**Reply by chuntaojun** [reply]
// 会后跟进处理

**Comment by andrewshan** [comment]
polaris原地替换eureka，需求讨论：https://github.com/polarismesh/polaris/issues/668
